mods/fire/init.lua

```diff
@@ -132,13 +132,13 @@ minetest.override_item("default:coalblock", {
 	after_destruct = function(pos, oldnode)
 		pos.y = pos.y + 1
 		if minetest.get_node(pos).name == "fire:permanent_flame" then
-			minetest.remove_node(pos)
+			minetest.swap_node(pos, {name = "air"})
```

Comment:
But this removed flame node may have metadata.

Reply:
When `after_destruct` is called, the metadata is already gone. See:
[`ServerEnvironment::setNode`](https://github.com/minetest/minetest/blob/24b6bd0f679f/src/serverenvironment.cpp#L948-L956) (node replacement, node metadata removal and `after_destruct` afterwards)
`Map::addNodeAndUpdate`
`Map::removeNodeMetadata` (metadata removal)